### PR TITLE
Add a flag to terminate workflows in case of failure

### PR DESCRIPTION
**Pull request by eapolinario**
This PR adds a flag to terminate workflow executions in case of a failure in the functional test. Such workflow executions are terminated and the reason is marked as "aborting execution scheduled in functional test", e.g.:

Screenshot from 2022-03-01 23-57-07

Here's an example of using this new flag:
```
> python functional-tests/run-tests.py $LATEST_VERSION P0,P1 ~/.flyte/config core True
```
Notice the string `True` at the end of the command line. 

The number of flags is unwieldy, for this reason a follow-up PR is going to put all of them under a more sane regime.

Signed-off-by: Eduardo Apolinario <eapolinario@users.noreply.github.com>

### CODE REVIEW

**Comment by eapolinario** [comment]
I tested this on a sandbox.

File: flytetester/functional-tests/run-tests.py [hunk 1]
```diff
@@ -180,8 +181,9 @@ def run(release_tag: str, priorities: List[str], config_file_path) -> List[Dict[
     priorities = sys.argv[2].split(',')
     config_file = sys.argv[3]
     return_non_zero_on_failure = sys.argv[4]
+    terminate_workflow_on_failure = True if sys.argv[5] == 'True' else False
```
Comment (evalsocket) [inline flytetester/functional-tests/run-tests.py:184]: Does it fail if we didn't pass argument by default ?

**Comment by evalsocket** [comment]
@eapolinario sorry for the confusion but we can remove this run script from flytetools, We have a common script in boilerplate https://github.com/flyteorg/boilerplate/tree/master/boilerplate/flyte/end2end

**Comment by eapolinario** [comment]
Sorry for the confusion. 

I'm going to close this PR and make the change in the `boilerplate`.

**Pull request closed without merging**